framework/src/modules/dynamic_reward/module.ts

```diff
@@ -280,9 +280,9 @@ export class DynamicRewardModule extends BaseModule {
 		if (!bftValidator) {
 			throw new Error('Invalid generator. Validator params does not include the validator.');
 		}
-		const stakeRewardActiveValidators = getStakeRewardActiveValidators(
+		const stakeRewardActiveValidators = await getStakeRewardActiveValidators(
 			context,
-			this._posMethod,
+			this._validatorMethod,
```

Comment:
Here I would just pass `numberActiveValidators` as an argument, as in the suggestion in the issue description. You can get `numberActiveValidators` from `validatorsParams.validators`, by counting the items with bft weight > 0. 

Is there any reason to do it differently?